File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int n=nums.size();
        int x=0,y=0;
        while(x<n && y<n){
            if(nums[x]==val) break;
            x++;y++;
        }
        while(x<n && y<n){
            while(x<n){
                if(nums[x]==val) break;
                else x++;
            }
            while(y<n){
                if(nums[y]!=val) break;
                else y++;
            }
            if(x<n && y<n)swap(nums[x],nums[y]);
        }
        int res=0;
        for(auto x:nums){
            if(x==val) break;
            res++;
        }
        return res;
    }
};

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        write = 0
        for read in range(0,len(nums)):
            if nums[write]!=val:
                write+=1
            else:
                if nums[read]!=val:
                    nums[write],nums[read]=nums[read],nums[write]
                    write+=1
        return write